src/main/java/org/commcare/session/CommCareSession.java

```diff
@@ -204,10 +205,16 @@ public String getNeededData(EvaluationContext evalContext) {
         } else if (entriesForCurrentCommand.isEmpty()) {
             // No entries available directly within the current command, so we must need to select another menu
             return SessionFrame.STATE_COMMAND_ID;
-        } else if (entriesForCurrentCommand.size() == 1
-                && entriesForCurrentCommand.elementAt(0).getPostRequest() != null
-                && entriesForCurrentCommand.elementAt(0).getPostRequest().isRelevant(evalContext)) {
-            return SessionFrame.STATE_SYNC_REQUEST;
+        } else if (entriesForCurrentCommand.size() == 1) {
+            Entry entry = getPlatform().getEntry(currentCmd);
+            if (entry == null) {
```

Comment:
this should never happen since we have already established `entriesForCurrentCommand.size() == 1` ?

Reply:
This can happen if the command is a menu command that only has one entry.

Reply:
I don't think I am understanding it correctly. Why would `entry` be null if the command menu has only one entry ?

Reply:
Because if the command is for a menu then it can't also be for an entry right? So `getPlatform().getEntry()` won't find an Entry with that command ID.

Reply:
yes, makes sense